examples/lock-app/silabs/src/LockManager.cpp

```diff
@@ -743,25 +879,79 @@ bool LockManager::setLockState(chip::EndpointId endpointId, const Nullable<chip:
         return false;
     }
 
-    // Check the PIN code
-    for (const auto & currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
+    // Check all pin codes associated to all users to see if this pin code exists
+    for (int userIndex = 0; userIndex < kMaxUsers; userIndex++)
     {
+        // Get user data to obtain currentCredentialCount
```

Comment:
Is there a reason to not just call GetUser here?

Reply:
Good point. Made this change